lib/dpl/provider/pages.rb

```diff
@@ -44,34 +63,134 @@ def slug
         options.fetch(:repo) { context.env['TRAVIS_REPO_SLUG'] }
       end
 
+      def keep_history
+        options.fetch(:keep_history, false)
+      end
+
+      def committer_from_gh
+        options.fetch(:committer_from_gh, false)
+      end
+
+      def allow_empty_commit
+        options.fetch(:allow_empty_commit, false)
+      end
+
+      def verbose
+        # Achtung! Never verbosify git, since it may expose user's token.
+        options.fetch(:verbose, false)
+      end
+
+      def api  # Borrowed from Releases provider
+        error 'gh-token must be provided for Pages provider to work.' unless @gh_token
+
+        @api ||= Octokit::Client.new(:access_token => @gh_token)
```

Comment:
(just a side note)
It looks I didn't take into account custom GitHub Entreprise URLs.
Need to check whether `@gh_url` could be bypassed in here.

Reply:
[done via #735]